lib/api/core/plugins/pluginsManager.js

```diff
@@ -845,15 +852,19 @@ function triggerHooks(emitter, event, data) {
  * @returns {Promise}
  */
 function triggerPipes(pipes, event, data) {
-  let preparedPipes = [];
-  const wildcardEvent = getWildcardEvent(event);
+  const preparedPipes = [];
+  const wildcardEvents = getWildcardEvents(event);
 
   if (pipes && pipes[event] && pipes[event].length) {
-    preparedPipes = pipes[event];
+    preparedPipes.push(...pipes[event]);
```

Comment:
I think the following is far better performances wise:

```suggestion
    preparedPipes = Array.from(pipes[event]);
```

Reply:
push x 10,197,616 ops/sec ±5.46% (68 runs sampled)
array from x 1,498,289 ops/sec ±2.96% (76 runs sampled)
Fastest is push

Reply:
:astonished: :+1: